obspy/io/seiscomp/tests/test_inventory.py

```diff
@@ -306,6 +306,17 @@ def test_stage_empty_poles_and_zeros(self):
         self.assertEqual(zeros, [])
         self.assertEqual(poles, [])
 
+    def test_omit_response(self):
+        """
+        Test SC3ML inventory with response information that leads to ObspyException
+        Read up to channel level and omit response information
+        """
+        inv = read_inventory(os.path.join(self.data_dir,
+                                          "IM.I31KZ.sc3ml"))
+        self.assertEqual(inv[0].code, "IM")
+        self.assertEqual(inv[0][0].code, "I31H1")
+        for cha in inv[0][0].channels:
+            self.assertTrue(cha.code in ["BDF", "LWD", "LWS", "LKO", "LDA"])
```

Comment:
This is not making sure all those channels are actually present.

```
expected = ["BDF", "LWD", "LWS", "LKO", "LDA"]
got = [cha.code for cha in inv[0][0].channels]
self.assertEqual(expected, got)
```

Ideally you also want to test that `.response` is `None` for the invalid ones and that there is a response for the valid ones.

Reply:
Added these lines as well as a check that the response stages are empty.